src/edgepi/adc/adc_voltage.py

```diff
@@ -69,9 +72,15 @@ def code_to_voltage(code: list[int], adc_info: ADCReadInfo, calibs: CalibParam)
     code_bits = bitstring_from_list(code[:adc_info.num_data_bytes])
     num_bits = adc_info.num_data_bytes * 8
     code_val = code_bits.uint
-    # handling negative number
-    if _is_negative_voltage(code_bits):
-        code_val = code_val - 2**num_bits
+
+    if _is_negative_voltage(code_bits) and adc_info.num_data_bytes == 4:
+        code_val = code_val - ADC1_UPPER_LIMIT
+    elif _is_negative_voltage(code_bits) and adc_info.num_data_bytes == 3:
+        code_val = code_val - ADC2_UPPER_LIMIT
+    elif adc_info.num_data_bytes == 4:
+        code_val = code_val + ADC1_UPPER_LIMIT
+    elif adc_info.num_data_bytes == 3:
+        code_val = code_val + ADC2_UPPER_LIMIT
```

Comment:
what do `adc_info.num_data_bytes == 4` and  `adc_info.num_data_bytes == 3` mean? is that how we identify adc1 and adc2 ? should we add comments to clarify?

Reply:
[0c5a3d0](https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/385/commits/0c5a3d08ee9674de0c0c474223d095c230d743dc)

I created a constants for these numbers